ts/Extensions/ExportData/ExportData.ts

```diff
@@ -1016,14 +1017,22 @@ function chartToggleDataTable(
 
     // Toggle the visibility
     if (this.dataTableDiv) {
+        const style = this.dataTableDiv.style,
+            oldDisplay = style.display;
 
-        this.dataTableDiv.style.display = show ? 'block' : 'none';
+        style.display = show ? 'block' : 'none';
 
         // Generate the data table
         if (show) {
             this.dataTableDiv.innerHTML = AST.emptyHTML;
             const ast = new AST([this.getTableAST()]);
             ast.addToDOM(this.dataTableDiv);
+
+            if (createContainer || oldDisplay !== style.display) {
+                // Was hidden, is shown
+                fireEvent(this, 'afterDataTableShown', this.dataTableDiv);
```

Comment:
Instead of adding another event that triggers immediately before the `afterViewData` event, maybe we can just extend the functionality of the original one? If we pass a parameter like `e.wasHidden` or similar, we can just check for that in the a11y handler.

Reply:
That's way smarter!